RFC-0006-nan-definition.md

```diff
@@ -0,0 +1,179 @@
+Big query does not have functions that do distincts or sorting on arrays or maps
+https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types 
+https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#max
+
+#### MySQL
+
+Mysql doesn't support NaNs at all (see reference here https://dev.mysql.com/doc/refman/8.0/en/type-conversion.html)
+
+#### Oracle
+Oracle orders NaN greatest with respect to all other values, and evaluates NaN equal to NaN.
+
+https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Data-Types.html#GUID-33A52FDB-BA5C-474E-96D3-40390BA5F5F4
+
+#### DB2
+
+DB2 has 2 kinds of floating point numbers.  The first, DOUBLE or FLOAT, does not support NaN or Infinity values.  The second, DECFLOAT, supports Infinities as well as quiet and signaling NaNs. It is described as an IEEE-754 number, but does not follow the IEEE spec for comparisons.  The NaNs compare as equal to each other and greater magnitude than any other number (The ordering among the different special values is as follows: -NAN < -SNAN < -INFINITY < 0 < INFINITY < SNAN <NAN)
+https://www.ibm.com/docs/en/db2-for-zos/13?topic=numbers-double-precision-floating-point-double-float
+https://www.ibm.com/docs/en/db2-for-zos/13?topic=numbers-decimal-floating-point-decfloat
+https://www.ibm.com/docs/en/db2-for-zos/13?topic=comparison-numeric-comparisons 
+
+
+## Goals
+
+* Define a consistent behavior for ordering and comparisons of Nans in Presto
+* Make behavior easy to implement and debug
+* Make behavior easy for users to understand
+* Align behavior with Velox for consistency across Java and C++ workers
+
+## Proposed Implementation
+
+We propose defining ordering and comparisons for NaNs such that `NaN = NaN` and sorts biggest.
+
+With this proposal we stray from the IEEE definition and follow the example of postgresql and others.  In this case for all equality purposes (including =, distinctness, grouping, join keys, etc.) we would define NaN=NaN.  We would sort NaN as greater than infinity for all purposes, including for min/max and > and < comparisons.
+
+This was by far the most common behavior across the DBS we looked at, including Spark, which at Meta is the engine we care most about compatibility with.  It is not IEEE-754 compliant, but no other DB we looked at was (either they did not support NaNs or they had at least some features where they strayed from IEEE compliance). And given how erratically we were behaving previously, it is safe to say that our users have not been relying on IEEE-754 compliance (or if they have, they certainly have not been getting it). This definition is consistent across all equality,comparison, and ordering operations meaning that users can always know what to expect, and though it would require some changes in Presto, the implementation should be pretty straight forward and reduces the chances of bugs. Velox has a proposal for the same: https://github.com/facebookincubator/velox/pull/7237.
```

Comment:
```suggestion
This was by far the most common behavior across the DBS we looked at, including Spark, which at Meta is the engine we care most about compatibility with.  It is not IEEE-754 compliant, but no other DB we looked at was (either they did not support NaNs or they had at least some features where they strayed from IEEE compliance). Given how erratically we behaved previously, it is safe to say that our users have not been relying on IEEE-754 compliance - if they have, they certainly have not been getting it. 

This definition is consistent across all equality, comparison, and ordering operations meaning that users can always know what to expect. Though it would require some changes in Presto, the implementation should be straightforward and reduces the chances of bugs. Velox has a proposal for the same: https://github.com/facebookincubator/velox/pull/7237.
```
Suggest breaking up this paragraph for easier reading. Also, consider replacing "straightforward" with a more common word to improve readability. Maybe "simple"?